src/outagesim/backup_reliability.jl

```diff
@@ -505,77 +769,217 @@ function backup_reliability_inputs(d::Dict, p::REoptInputs; r::Dict = Dict())::D
+
+# Arguments
+- `r::Dict`: Dictionary of inputs for reliability calculations.
+    inputs of r:
+    -critical_loads_kw::Array                  Critical loads per time step. (Required input)
+    -microgrid_only::Bool = false              Boolean to specify if only microgrid upgraded technologies run during grid outage 
+    -chp_capacity::Real                        CHP capacity. 
+    -pv_size_kw::Real                          Size of PV System
+    -pv_production_factor::Array               PV production factor per time step (required if pv_size_kw in dictionary)
+    -pv_migrogrid_upgraded::Bool               If true then PV runs during outage if microgrid_only = TRUE (defaults to false)
+    -battery_size_kw::Real                     Battery capacity. If no battery installed then PV disconnects from system during outage
+    -battery_size_kwh::Real                    Battery energy storage capacity
+    -charge_efficiency::Real                   Battery charge efficiency
+    -discharge_efficiency::Real                Battery discharge efficiency
+    -battery_year_one_soc_series_pct           Battery state of charge in each hour (if not input then defaults to battery size)
+    -generator_failure_to_start::Real = 0.0066                Chance of generator starting given outage
+    -generator_failure_to_run::Real = 0.00157                 Chance of generator failing in each hour of outage
+    -num_generators::Int = 1                                  Number of generators. Will be determined by code if set to 0 and gen capacity > 0.1
+    -generator_size_kw::Real = 0.0                           Backup generator capacity. Will be determined by REopt optimization if set less than 0.1
+    -num_battery_bins::Int = 101                        Internal value for discretely modeling battery state of charge
+    -max_outage_duration::Int = 96                      Maximum outage hour modeled
+```
+"""
+function backup_reliability_inputs(;r::Dict)::Dict
+    
+    
+    r2 = Dict()
+    r2[:generator_operational_availability] = get(r, "generator_operational_availability", 0.9998)
+    r2[:generator_failure_to_start] = get(r, "generator_failure_to_start", 0.0066)
+    r2[:generator_failure_to_run] = get(r, "generator_failure_to_run", 0.00157)
+    r2[:num_generators] = get(r, "num_generators", 1)
+    r2[:generator_size_kw] = get(r, "generator_size_kw", 0.0)
+    r2[:max_outage_duration] = get(r, "num_battery_bins", 96)
+    
+    net_critical_loads_kw = r["critical_loads_kw"]
+    zero_array = zeros(length(net_critical_loads_kw))
+
+    if "chp_capacity" in keys(r)
+        net_critical_loads_kw .-= r["chp_capacity"]
+    end
+
+    microgrid_only = get(r, "microgrid_only", false)
+
+    pv_kw_ac_hourly = zero_array
+    if "pv_size_kw" in keys(r)
+        pv_kw_ac_hourly = r["pv_size_kw"] .* r["pv_production_factor_series"]
+    end
+    if microgrid_only && !Bool(get(r, "pv_migrogrid_upgraded", false))
```

Comment:
I don't think pv_kw_ac_hourly should be set to zero in this case if the user provided a pv_size_kw to overwrite the REopt size. Actually I don't really see the need for these microgrid only inputs at all. Seems like an unnecessary complication. If the user doesn't want to include PV in the calculations, they can just provide pv_size_kw = 0.

Reply:
I could see microgrid_only being used in backup_reliability_reopt_inputs along with the REopt decision variables of whether PV and battery are upgraded, but it doesn't make sense to have in backup_reliability_inputs